storage/pebble.go

```diff
@@ -3,13 +3,44 @@ package storage
 import (
 	"errors"
 	"fmt"
+	"math"
 
 	"github.com/cockroachdb/pebble"
 	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"go.uber.org/multierr"
 
 	storageErrors "github.com/gnolang/tx-indexer/storage/errors"
 )
 
+const (
+	// keyLatestHeight is the quick lookup key
+	// for the latest height saved in the DB
+	keyLatestHeight = "/meta/lh"
+
+	// keyBlocks is the key for each block saved. They are stored by height
+	prefixKeyBlocks = "/data/blocks/"
+
+	// keyTxs is the prefix for each transaction saved.
+	prefixKeyTxs = "/data/txs/"
+)
+
+func keyTx(blockNum int64, txIndex uint32) []byte {
+	var key []byte
```

Comment:
Especially if we change varint to just using `uint64`, the resulting values from `keyTx` and `keyBlock` are constant sized. Consider pre-allocating the slice (`make([]byte, 0, 8 + 4 + ...)`).

Reply:
I tried to do that optimization, but the size of the string serialization is not known here.